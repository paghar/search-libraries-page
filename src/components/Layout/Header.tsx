import Search from "@/components/misc/svg/Search";
import Button from "@/components/misc/Button";
import Textbox from "@/components/misc/Textbox";
import {SearchAction} from "@/utils/enum/search";

interface IProps{   
  searchText:string
  dispatch:any
  searchClick:()=>void
}

const Header = ({searchText,dispatch,searchClick}:IProps) => {

  const borderRadiusStyle = {
    borderBottomLeftRadius: "50% 20%",
    borderBottomRightRadius: "50% 20%",
  };

  return(    
    <div className="overflow-hidden relative flex items-center min-h-32">  
      <div style={borderRadiusStyle} className="absolute bg-pink-700 w-full h-1/2 z-0 top-0"/>
      <div className="absolute inset-x-auto w-full z-10">
        <div className="w-2/3 mx-auto shadow-md rounded-md p-4 bg-white">           
          <form className="flex items-center"> 
            <div className="relative w-full">                
              <div className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none">
                <Search className={`"w-5 h-5 text-gray-500 dark:text-gray-400`}/>            
              </div>
              <Textbox
                id="search"                
                value={searchText}
                type="text"
                placeholder="Search ..."
                onChange={(e) => dispatch({type: SearchAction.SetSearchText,payload:e.target.value})} 
              />      
            </div>
            <Button
              id="search"
              type="button"             
              onClick={searchClick}
              text="Search"
            >
              <Search className={`mr-2 -ml-1 w-5 h-5`}/>
            </Button>                     
          </form>         
        </div>
      </div>
    </div>
  );
};

export default Header;