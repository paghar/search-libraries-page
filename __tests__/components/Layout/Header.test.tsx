import React from "react";
import {render, fireEvent} from "@testing-library/react";
import Header from "../../../src/components/Layout/Header";

const searchText = "Test search";
const dispatch = jest.fn();
const searchClick = jest.fn();

describe("Header Component", () => {
  
  const {container} = render(
    <Header searchText={searchText} dispatch={dispatch} searchClick={searchClick} />
  );
  
  it("Renders Snapshot Header", () => {    
    expect(container).toMatchSnapshot();
  });
  
});

describe("Header Component", () => {
  test("renders header with search input and button", () => {
        
    const {getByPlaceholderText, getByText} = render(
      <Header searchText={searchText} dispatch={dispatch} searchClick={searchClick} />
    );

    const searchInput = getByPlaceholderText("Search ...");
    expect(searchInput).toBeInTheDocument();
    expect(searchInput).toHaveValue(searchText);

    const searchButton = getByText("Search");
    expect(searchButton).toBeInTheDocument();
    fireEvent.click(searchButton);
    expect(searchClick).toHaveBeenCalled();
  });
});