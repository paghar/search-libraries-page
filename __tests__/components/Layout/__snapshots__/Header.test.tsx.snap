// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Renders Snapshot Header 1`] = `
<div>
  <div
    class="overflow-hidden relative flex items-center min-h-32 w-full"
  >
    <div
      class="absolute bg-pink-700 w-full h-1/2 z-0 top-0"
      style="border-bottom-left-radius: 50% 20%; border-bottom-right-radius: 50% 20%;"
    />
    <div
      class="absolute inset-x-auto w-full z-10"
    >
      <div
        class="lg:w-2/5 md:w-2/5 max-sm:w-full mx-auto shadow-md rounded-md p-4 bg-white"
      >
        <form
          class="flex items-center justify-evenly lg:flex-nowrap md:flex-wrap max-sm:flex-wrap "
        >
          <div
            class="relative lg:w-5/6 md:w-full md:mb-2 max-sm:w-full max-sm:mb-2"
          >
            <div
              class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
            >
              <svg
                class=""w-5 h-5 text-gray-500 dark:text-gray-400"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <input
              class="   bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg   focus:ring-pink-500 focus:border-pink-500 block w-full pl-10 p-2.5    dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400    dark:text-white dark:focus:ring-pink-500 dark:focus:border-pink-500   "
              id="search"
              placeholder="Search ..."
              type="text"
              value="Test search"
            />
          </div>
          <button
            class="w-28 
        inline-flex items-center 
        py-2.5 px-3 ml-2 text-sm 
        font-medium text-white
        bg-pink-700 rounded-lg border border-pink-700
        hover:bg-pink-800 focus:ring-4 
        focus:outline-none focus:ring-pink-300
        dark:bg-pink-600 dark:hover:bg-pink-700 dark:focus:ring-pink-800
      "
            id="search"
            type="button"
          >
            <svg
              class="mr-2 -ml-1 w-5 h-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Search
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
